File: sdks/sdkwork-craw-chat-sdk/sdkwork-craw-chat-sdk-typescript/composed/src/streams-module.ts
import { buildTextFrameRequest } from './builders.js';
import type {
  AbortStreamRequest,
  AppendStreamFrameRequest,
  AppendTextFrameOptions,
  CheckpointStreamRequest,
  CompleteStreamRequest,
  OpenStreamRequest,
  QueryParams,
  StreamFrame,
  StreamFrameWindow,
  StreamSession,
} from './types.js';
import type { CrawChatSdkContext } from './sdk-context.js';

export class CrawChatStreamsModule {
  constructor(private readonly context: CrawChatSdkContext) {}

  open(body: OpenStreamRequest): Promise<StreamSession> {
    return this.context.backendClient.stream.open(body);
  }

  listFrames(
    streamId: string | number,
    params?: QueryParams,
  ): Promise<StreamFrameWindow> {
    return this.context.backendClient.stream.listStreamFrames(streamId, params);
  }

  appendFrame(
    streamId: string | number,
    body: AppendStreamFrameRequest,
  ): Promise<StreamFrame> {
    return this.context.backendClient.stream.appendStreamFrame(streamId, body);
  }

  appendTextFrame(
    streamId: string | number,
    options: AppendTextFrameOptions,
  ): Promise<StreamFrame> {
    return this.appendFrame(streamId, buildTextFrameRequest(options));
  }

  checkpoint(
    streamId: string | number,
    body: CheckpointStreamRequest,
  ): Promise<StreamSession> {
    return this.context.backendClient.stream.checkpoint(streamId, body);
  }

  complete(
    streamId: string | number,
    body: CompleteStreamRequest,
  ): Promise<StreamSession> {
    return this.context.backendClient.stream.complete(streamId, body);
  }

  abort(
    streamId: string | number,
    body: AbortStreamRequest,
  ): Promise<StreamSession> {
    return this.context.backendClient.stream.abort(streamId, body);
  }
}

File: sdks/sdkwork-craw-chat-sdk/sdkwork-craw-chat-sdk-flutter/composed/lib/src/streams_module.dart
import 'package:backend_sdk/backend_sdk.dart';

import 'builders.dart';
import 'context.dart';
import 'types.dart';

class CrawChatStreamsModule {
  final CrawChatSdkContext context;

  CrawChatStreamsModule(this.context);

  Future<StreamSession?> open(OpenStreamRequest body) {
    return context.backendClient.stream.open(body);
  }

  Future<StreamFrameWindow?> listFrames(
    String streamId, [
    CrawChatQueryParams? params,
  ]) {
    return context.backendClient.stream.listStreamFrames(streamId, params);
  }

  Future<StreamFrame?> appendFrame(
    String streamId,
    AppendStreamFrameRequest body,
  ) {
    return context.backendClient.stream.appendStreamFrame(streamId, body);
  }

  Future<StreamFrame?> appendTextFrame(
    String streamId,
    CrawChatAppendTextFrameOptions options,
  ) {
    return appendFrame(
      streamId,
      CrawChatBuilders.textFrame(options),
    );
  }

  Future<StreamSession?> checkpoint(
    String streamId,
    CheckpointStreamRequest body,
  ) {
    return context.backendClient.stream.checkpoint(streamId, body);
  }

  Future<StreamSession?> complete(
    String streamId,
    CompleteStreamRequest body,
  ) {
    return context.backendClient.stream.complete(streamId, body);
  }

  Future<StreamSession?> abort(
    String streamId,
    AbortStreamRequest body,
  ) {
    return context.backendClient.stream.abort(streamId, body);
  }
}
